### PR TITLE
cli: web advertise exit node button

### PR DESCRIPTION
**Pull request by crawshaw**
A couple of gnarly assumptions in this code, as always with the async
message thing.

UI button is based on the DNS settings in the admin panel.

Co-authored-by: Maisem Ali <maisem@tailscale.com>
Signed-off-by: David Crawshaw <crawshaw@tailscale.com>

### CODE REVIEW

**Comment by bradfitz** [comment]
I'm happy to defer to @maisem on this review :)

File: cmd/tailscale/cli/web.html [hunk 1]
```diff
@@ -86,14 +86,28 @@ <h3 class="text-3xl font-semibold mb-3">Log in</h3>
...
+		<a href="#" class="mb-4 link font-medium js-loginButton" target="_blank">Reauthenticate</a>
+	</div>
 	{{ end }}
 </main>
 <script>(function () {
-let loginButtons = document.querySelectorAll(".js-loginButton");
+const advertiseExitNode = {{.AdvertiseExitNode}};
 let fetchingUrl = false;
+var data = {
+	AdvertiseRoutes: "{{.AdvertiseRoutes}}",
```
Comment (maisem) [inline cmd/tailscale/cli/web.html:105]: Maybe remove this as it's not being used?

File: cmd/tailscale/cli/web.go [hunk 2]
```diff
@@ -49,11 +51,13 @@ func init() {
...
-	Status       string
-	DeviceName   string
-	IP           string
+	Profile           tailcfg.UserProfile
+	SynologyUser      string
+	Status            string
+	DeviceName        string
+	IP                string
+	AdvertiseExitNode bool
+	AdvertiseRoutes   string
```
Comment (maisem) [inline cmd/tailscale/cli/web.go:60]: Remove this, it's not being used

**Review by maisem — APPROVED** [review]
I could not figure out how this worked until I actually ran the code :)

File: cmd/tailscale/cli/web.html (hunk 1 above)
Comment (crawshaw) [reply cmd/tailscale/cli/web.html:105]: I tried this, but it involves unwinding a stack of other code and I'd rather not, as all of this is neatly in place to support advertising subnet routes. At this point all the "server" code is done and all that's needed on top of this PR is a UI.

**Comment by crawshaw** [comment]
Updated spacing and button wording as per @sonovawolf's advice.

**Pull request merged**